src/analytics.js

```diff
@@ -1,34 +1,28 @@
-const fs = require("fs")
-const path = require('path')
-
-const config = require('./config');
-
+const path = require("path")
+const config = require('./config')
 const GoogleAnalyticsClient = require("./google-analytics/client")
 const GoogleAnalyticsDataProcessor = require("./process-results/ga-data-processor")
 
-// The reports we want to run.
-var reports_path = config.reports_file || (path.join(process.cwd(), "reports/reports.json"));
-var reports = JSON.parse(fs.readFileSync(reports_path)).reports;
-var by_name = {};
-for (var i=0; i<reports.length; i++)
-    by_name[reports[i].name] = reports[i];
-
-// Google Analytics data fetching and transformation utilities.
-// This should really move to its own analytics.js file.
-var Analytics = {
-
-    reports: by_name,
-
-    query: function(report, callback) {
-        if (!report) {
-            return callback()
-        }
-
-        return GoogleAnalyticsClient.fetchData(report).then(data => {
-            const processedData = GoogleAnalyticsDataProcessor.processData(report, data)
-            callback(null, processedData)
-        }).catch(callback)
-    },
-};
-
-module.exports = Analytics;
+const query = (report) => {
+  if (!report) {
+    return Promise.reject("Analytics.query missing required argument `report`")
+  }
+
+  return GoogleAnalyticsClient.fetchData(report).then(data => {
+    return GoogleAnalyticsDataProcessor.processData(report, data)
+  })
+}
+
+const _loadReports = () => {
+  const reports = {}
+  const _reportFilePath = path.resolve(process.cwd(), config.reports_file || "reports/reports.json")
```

Comment:
We were actually using `path` to construct the filepaths before. We were only using `fs` to load the contents of the reports file.

One change that may be confusing is we were using `path.join`. I replaced that with `path.resolve` since `require` resolves paths relative to the module instead of the current working directory. Using `path.resolve()` will gives us an absolute path. If `config.reports_file` is a relative path, it will be resolved relative to the cwd. If it is an absolute path, it will not be changed. To my knowledge, this should maintain the existing behavior.